File: java/src/test/java/problems1_100/MinimumWindowSubstringTest.java
package problems1_100;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class MinimumWindowSubstringTest extends SampledTest {
    @Test
    public void test() {
        for (MinimumWindowSubstring solution : MinimumWindowSubstring.values()) {
            for (MinimumWindowSubstringSample sample
                : getSamples(MinimumWindowSubstringSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.minWindow(sample.input.s, sample.input.t))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/MinimumWindowSubstringTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class MinimumWindowSubstringTest : SampledTest() {
    @Test
    fun test() =
        MinimumWindowSubstring.entries.forEach { solution ->
            getSamples<Array<MinimumWindowSubstringSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.minWindow(input.s, input.t))
                    .isEqualTo(output)
            }
        }
}
